synapse/storage/databases/main/room.py

```diff
@@ -699,20 +699,30 @@ def delete_ratelimit_txn(txn: LoggingTransaction) -> None:
         await self.db_pool.runInteraction("delete_ratelimit", delete_ratelimit_txn)
 
     @cached()
-    async def get_retention_policy_for_room(self, room_id: str) -> Dict[str, int]:
+    async def get_retention_policy_for_room(
+        self, room_id: str
+    ) -> Dict[str, Optional[int]]:
```

Comment:
Not asking for a change, but I'd have more confidence in the code using this if it returned an attrs with the two fields.

Reply:
That's a good point.

Reply:
Looks much clearer to me! thank you!